src/distributions/other.rs

```diff
@@ -161,28 +165,43 @@ impl<T> Distribution<Option<T>> for Uniform where Uniform: Distribution<T> {
 #[cfg(test)]
 mod tests {
     use {Rng, RngCore, Uniform};
+    use distributions::Alphanumeric;
     #[cfg(all(not(feature="std"), feature="alloc"))] use alloc::String;
-    
+
     #[test]
     fn test_misc() {
         let rng: &mut RngCore = &mut ::test::rng(820);
         
         rng.sample::<char, _>(Uniform);
         rng.sample::<bool, _>(Uniform);
     }
-    
+
     #[cfg(any(feature="std", feature="alloc"))]
     #[test]
     fn test_chars() {
         use core::iter;
-        use distributions::Alphanumeric;
         let mut rng = ::test::rng(805);
-        
-        let c = rng.sample(Alphanumeric);
-        assert!((c >= '0' && c <= '9') || (c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z'));
-        
+
+        // Test by generating a relatively large number of chars, so we also
+        // take the rejection sampling path.
         let word: String = iter::repeat(())
-                .map(|()| rng.sample(Alphanumeric)).take(5).collect();
-        assert_eq!(word.len(), 5);
+                .map(|()| rng.gen::<char>()).take(1000).collect();
+        assert!(word.len() != 0);
```

Comment:
There's not much point keeping this now. There's already a doc-test on `Rng` doing the same thing.

Reply:
I can't find that doc-test to be honest.

Reply:
```
$ rg Alphanumeric src/lib.rs 
309:/// use rand::distributions::{Alphanumeric, Range};
318:///         .map(|()| rng.sample(Alphanumeric))
...
```

Reply:
I see the difference :smile:. I was looking for a doctest for `.gen::char()`. I would like to keep the test for `.sample(Alphanumeric)`, because it checks wheter the characters are actually alphanumeric.

Reply:
Yes, the new one you added does, but the one above only makes a `String`, which seems redundant.

Reply:
Yes, the one in `lib.rs` is redundant as a test. But it is still useful as documentation, isn't it?

Reply:
Yes, the one in `lib.rs` is redundant with `test_chars` which is pointless now you added `test_alphanumeric`. I'm saying delete `test_chars`.

Reply:
But `test_chars` is testing the `Uniform` distributions for chars, not the `Alphanumeric` one like `test_alphanumeric` and the doc-test?

Reply:
Oh, you changed it. I guess I should read not just rely on memory. :+1: